docs/source/high-level-graphs.rst

```diff
@@ -0,0 +1,193 @@
+High Level Graphs
+=================
+
+Dask graphs produced by collections like Arrays, Bags, and DataFrames have
+high-level structure that can be useful for visualization and high-level
+optimization.  The task graphs produced by these collections encode this
+structure explicitly as ``HighLevelGraph`` objects.  This document describes
+how to work with these in more detail.
+
+
+Motivation and Example
+----------------------
+
+In full generality, Dask schedulers expect arbitrary task graphs, where each
+node is a single Python function call, and each edge is a dependency between
+two function calls.  These are usually stored in flat dictionaries.  Here is
+some simple dask dataframe code and the task graph that it might generate:
+
+
+.. code-block:: python
+
+    import dask.dataframe as dd
+
+    df = dd.read_csv('myfile.*.csv')
+    df = df + 100
+    df = df[df.name == 'Alice']
+
+.. code-block:: python
+
+   {
+    ('read-csv', 0): (pandas.read_csv, 'myfile.0.csv'),
+    ('read-csv', 1): (pandas.read_csv, 'myfile.1.csv'),
+    ('read-csv', 2): (pandas.read_csv, 'myfile.2.csv'),
+    ('read-csv', 3): (pandas.read_csv, 'myfile.3.csv'),
+    ('add', 0): (operator.add, 'myfile.0.csv', 100),
+    ('add', 1): (operator.add, 'myfile.1.csv', 100),
+    ('add', 2): (operator.add, 'myfile.2.csv', 100),
+    ('add', 3): (operator.add, 'myfile.3.csv', 100),
```

Comment:
As in the `HighLevelGraph` docstring, guessing `'myfile.*.csv'` should be `('read-csv', *)`.